pkg/beacon/relay/registry/groups.go

```diff
@@ -83,7 +84,7 @@ func (g *Groups) GetGroup(groupPublicKey []byte) []*Membership {
 // a new operation and it cannot have an ongoing operation for which it could be
 // selected before it expired. Such a group can be safely removed from the registry
 // and archived in the underlying storage.
-func (g *Groups) UnregisterStaleGroups() {
+func (g *Groups) UnregisterStaleGroups(latestGroupPublicKey []byte) {
```

Comment:
This change makes me wonder whether we can optimize the number of calls to Ethereum client by:
- making `UnregistarStaleGroups` private (this parameter looks a little weird now)
- calling `unregisterStaleGroups` from `RegisterGroup` (now this parameter does not look weird at all).

We don't need to clean up the storage every time a new group is registered, especially if the client has relatlively low stake and do not participate in a lot of new groups. We can avoid an excessive number of `IsStale` calls this way.

Reply:
I'm confused a bit about this part "We don't need to clean up the storage every time a new group is registered..", but by calling `groups.unregisterStaleGroups()` from `groups.RegisterGroup()` this is exactly what we'll do, won't we?

Reply:
Yes, but only for the current client, not in a global context. If you have just a min stake and you are qualifying to e.g. just 5% of new groups (random number, does not reflect the reality), you don't need to execute is stale check all the time, for 95% of DKGs, it's enough if you do it only when you qualified to a new group.

Reply:
Sooo... this suggestion wasn't ideal as I just realized `RegisterGroup` is called for every member of the group so for highly staked nodes we increased the number of calls instead of reducing it. I think we should revert 7371160 and do it in the way originally proposed here 😬

Reply:
😬 done